### PR TITLE
Add feature from #22769

### PR DESCRIPTION
**Pull request by Lixire**
- `Ctrl+A` now selects all files when focused on the scm viewlet

### CODE REVIEW

File: src/vs/workbench/electron-browser/commands.ts [hunk 1]
```diff
@@ -290,6 +290,25 @@ export function registerCommands(): void {
...
+		primary: KeyMod.CtrlCmd | KeyCode.KEY_A,
+		handler: (accessor) => {
+			const listService = accessor.get(IListService);
+			const focused = listService.getFocused();
+
+			if (focused instanceof List) {
+				const list = focused;
+				if (list.length !== 0) {
+					list.setSelection(Array.apply(null, { length: list.length }).map(Function.call, Number));
+					list.focusFirst();
```
Comment (roblourens) [inline src/vs/workbench/electron-browser/commands.ts]: Can we keep the same element focused?

File: src/vs/workbench/electron-browser/commands.ts [hunk 2]
```diff
@@ -290,6 +290,25 @@ export function registerCommands(): void {
...
+		when: ListFocusContext,
+		primary: KeyMod.CtrlCmd | KeyCode.KEY_A,
+		handler: (accessor) => {
+			const listService = accessor.get(IListService);
+			const focused = listService.getFocused();
+
+			if (focused instanceof List) {
+				const list = focused;
+				if (list.length !== 0) {
+					list.setSelection(Array.apply(null, { length: list.length }).map(Function.call, Number));
```
Comment (roblourens) [inline src/vs/workbench/electron-browser/commands.ts]: We have a helper for creating arrays over a range of numbers, see `range` in src/vs/base/common/arrays.ts.

**Review by roblourens — CHANGES_REQUESTED** [review]

**Review by bpasero — CHANGES_REQUESTED** [review]
Having a generic command for this that is being hooked up for any list/tree that is active does not work for me for the following reasons:
* the tree does not support multi select in most cases and now we are stealing `Ctrl+A` away from users. e.g. you can no longer select all in the editor when the explorer has focus
* the SCM viewlet should be the one that provides this command so that it only works there and it should also be smarter and not just blindly select all groups, but actually select all files of a group. today it will select also the group-categories (e.g. "Changes") which imho is not right.

image

**Comment by Lixire** [comment]
I agree with your input and was wondering how I can change the SCM viewlet to listen for a `ctrl + a` since it looks like keyboard support is set to false. The other (SCM exclusive) keyboard input it uses is coupled with a mouse click.

**Pull request closed without merging**